### ISSUE TITLE
Local time used where UTC should be

### ISSUE THREAD

**Issue by timadamson** [aws/aws-sdk-net-extensions-cognito#21]
Local time is used to keep track of timestamps when UTC should be used, otherwise time updates such as daylight savings shifts will cause errors.

```C#
        private CognitoUserSession GetCognitoUserSession(AuthenticationResultType authResult, string refreshTokenOverride = null)
        {
            string idToken = authResult.IdToken;
            string accessToken = authResult.AccessToken;
            string refreshToken;
            DateTime currentTime = DateTime.Now;
```
awssdk.extensions.cognitoauthentication\cognitouser.cs

```C#
        /// <summary>
        /// Determines if the tokens for a user are still valid
        /// </summary>
        /// <returns>Returns a boolean whether the user's tokens are still valid</returns>
        public bool IsValid()
        {
            DateTime currentTimeStamp = DateTime.Now;
```
awssdk.extensions.cognitoauthentication\cognitouser.cs

**Comment by todor-mutafov** [comment]
This issue persists.

Serializing the session data to a cookie converts all dates to UTC. This means that by the time you're done with a round-trip with the client, your invariant is already broken, from that point on the dates are in fact in UTC, and `IsValid` always fails because it's comparing with local time.

This also means that you can never ever run a `RefreshSignIn` call because the code checks if the session is valid before using the refresh token, and `IsValid` (as stated before) always returns `false`.

Are you gonna be fixing it?

**Comment by duaneking** [comment]
The use of non-utc time means that a server in one time zone can issue already expired keys in another time zone, right?

So when will this be fixed?

**Comment by ashishdhingra** [comment]
Fixed in version `2.2.1` as part of commit https://github.com/aws/aws-sdk-net-extensions-cognito/commit/ed78740725b1dd68ef9e76224501571bcdf54a0b

**Comment by duaneking** [comment]
I should not have had to point that out to get this fixed.
